packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -462,6 +462,23 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     return result;
   }
 
+  public async mineEmptyBlocks(
+    blockCount: BN = new BN(1),
+    intervalInSeconds: BN = new BN(1)
+  ) {
+    // first mine any pending transactions:
+    while (this._txPool.hasPendingTransactions()) {
+      await this.mineBlock();
+    }
+
+    const latestBlockNumber = await this.getLatestBlockNumber();
+    this._blockchain.addEmptyBlockRange({
+      first: latestBlockNumber,
+      last: latestBlockNumber.add(blockCount),
```

Comment:
```suggestion
      last: latestBlockNumber.add(blockCount).subn(mined),
```

also should take the amount of mined blocks so far.

Also, if mined === blockCount this part should be skipped.

Reply:
> also should take the amount of mined blocks so far.

Addressed in https://github.com/nomiclabs/hardhat/pull/2032/commits/4c6ad94fd0b602827f10565a120bbb58af4117e4

> Also, if mined === blockCount this part should be skipped.

Addressed in https://github.com/nomiclabs/hardhat/pull/2032/commits/4ceab4a641349ad7581e2caff86b970b9ca89080